cosmos/dbt/graph.py

```diff
@@ -635,6 +641,59 @@ def run_dbt_deps(self, dbt_cmd: str, dbt_project_path: Path, env: dict[str, str]
+        processing in the DAG run.
+        * Some users prefer to run `dbt deps` in the CI and "cache it" so Cosmos never runs `dbt deps`
+        * Some users want to use the "cached" dbt deps - but they would also like Cosmos to refresh the dependencies,
+        so they don't need to deploy Cosmos again.
+        * From an operating system perspective, to copy files takes more time than to create symbolic links.
+
+        Also, historically:
+        * Cosmos creates symbolic links to files/folders that are not updated by users, since this is compatible with read-only
+        dbt project paths and this is cheaper than copying those folders.
+
+        The current settings make sense for Cosmos 1.x and allow users to set things in different ways, while being backwards
+        compatible. We should review this for Cosmos 2.x.
+        """
+
+        should_not_create_dbt_deps_symbolic_link = self.should_install_dbt_deps or self.project.copy_dbt_packages
+
+        # The value of ignore_dbt_packages tells the function `create_symlinks` that we should not create a symbolic
+        # link for the `dbt_packages` folder. This can be desired in one or more of the two circumstances:
+        # 1. If we want to freshly install dbt packages (install_dbt_deps = True)
+        # 2. If we want to copy the dbt_packages folder instead of creating a symbolic link (copy_dbt_packages = True)
+        #
+        #  | Use case  | install_dbt_deps | copy_dbt_packages | create_symlinks.ignore_dbt_packages | what happens                 |
+        #  | A         | False            | False             | False                               | should create symlink        |
+        #  | B         | True             | False             | True                                | should run `dbt deps`        |
+        #  | C         | False            | True              | True                                | should copy dbt deps files   |
+        #  | D         | True             | True              | True                                | should copy & run `dbt deps` |
+        #
+        # Use cases description:
+        # A. High performance and deps may become outdated: Users run `dbt deps` outside of Cosmos and give pre-generated dbt packages. Dependencies may become outdated.
+        # B. Low performance and up-to-date deps: Cosmos always run `dbt deps` from scratch, every time the DAG is parsed (every time the DAG is parsed).
+        # C. (Non-practical) Middle performance and deps may become outdated: Users manage `dbt deps` outside of Cosmos and give pre-generated dbt packages. Dependencies may become outdated. More expensive than A, similar behaviour.
+        # D. Middle performance and up-to-date deps: Users run `dbt deps` outside of Cosmos and give pre-generated dbt packages. Cosmos runs dbt deps taking into account those user-generated files.
```

Comment:
I am trying to understand more here "Cosmos runs dbt deps taking into account those user-generated files."

What happens here? Do we or dbt take into account the version/checksum to handle installation of packages incrementally?

Reply:
I think If `dbt_packages` is missing, it will use `package-lock.yml` (if present) to install the correct package versions.